Bkong/1,2,3 data display

briannk opened the pull request:
Since I had two branches but this branch contains everything from the other branch, consider this branch to be the culmination of both. Am I intended to manually handle branches that split off a bit from each other? If branch 1 goes A (initial pull) -> B -> C, then branch 2 goes A -> Y -> Z with each letter denoting differing code, how should I approach resolving the potential conflicts when it comes time to merge? Say I want the branch to look like A -> B -> Z, how would I merge the branches so that I incorporate the desired changes (feature B and Z) but discard the rest (feature C and Y)?

briannk committed:
Created log-in/sign-up forms

briannk committed:
Added password reset functionality

briannk committed:
Created forms, though diagrams and css is still WIP

briannk committed:
Routing corrections

briannk commented:
hello this is bryan typing from brian's account. note to self: LGTM 👍

BryanHo10 commented on frontend/src/App.js:
```diff
@@ -0,0 +1,65 @@
+    <Router>
+      <Navbar />
+      <Routes>
+        <Route
```
It might look a bit cleaner to implement something like this:
https://stackoverflow.com/a/48497783

BryanHo10 commented on frontend/src/components/Dashboard.js:
```diff
@@ -0,0 +1,19 @@
+import Income from "./Income";
+
+const Dashboard = () => {
+  const containerStyles = `container mx-auto`;
```
You can pull this constant outside of the component, so that it doesn't have to reload every time the component is loaded.

BryanHo10 replied:
Do this likewise in other areas of the repo :)

briannk replied:
Should I extract the css variables to a new js module since I reuse this containerStyles variable a few times or do you mean to pull it outside of the Dashboard object? Does this make it a global variable?

BryanHo10 replied:
You could extract it into a new js module or just pull it out of the Dashboard object. There won't be a need to export container styles, so it's a variable isolated to this file.

BryanHo10 commented on frontend/src/components/DataViz.js:
```diff
@@ -0,0 +1,32 @@
+
+  const displayVisual = () => {
+    switch (chart) {
+      case "pie":
```
I would recommend creating a js enum-like object containing all of the different chart types:
ex)
```
const CHART_TYPE = {
...
PIE : 1,
LINE : 2,
...
}
```
This will make all CHART_TYPE.PIE consistent across application. 
Avoid passing in strings as a type as spelling mistakes are easily made and hard to track.

BryanHo10 commented on frontend/src/components/DataViz.js:
```diff
@@ -0,0 +1,32 @@
+      case "line":
+        return <LineViz dataProp={data} />;
+      default:
+        console.log("switch default");
```
Make sure the master branch is clean from console logs. Commenting out is fine if they're needed at a later time.

BryanHo10 commented on frontend/src/components/ExpenseForm.js:
```diff
@@ -0,0 +1,169 @@
+import { Button, Form, Message, Icon } from "semantic-ui-react";
+
+const ExpenseForm = () => {
+  const [expense, setExpense] = useState({
```
I won't comment too much on this because I'll suggest a library to use in the future that will clean up the form fields/validation implementation. 

This isn't the most scalable approach, but this is good for now :)

BryanHo10 commented on frontend/src/components/Expenses.js:
```diff
@@ -0,0 +1,18 @@
+     rounded p-4 mx-1`;
+  return (
+    <div className={containerStyles}>
+      <h2>{`Expenses from ${spendingData.startDate} to ${spendingData.endDate}`}</h2>
```
No need to interpolate the whole content of  h2. Take advantage of the JSX freedom, 
Ex]
```
<h2>Expenses from {spendingData.startDate} to {spendingData.endDate}</h2>
```

BryanHo10 commented on frontend/src/components/Income.js:
```diff
@@ -0,0 +1,18 @@
+     rounded p-4 mx-1`;
+  return (
+    <div className={containerStyles}>
+      <h2>{`Income from ${spendingData.startDate} to ${spendingData.endDate}`}</h2>
```
Same here:
No need to interpolate the whole content of  h2. Take advantage of the JSX freedom, 
Ex]
```
<h2>Income from {spendingData.startDate} to {spendingData.endDate}</h2>
```

BryanHo10 commented on frontend/src/components/LogIn.js:
```diff
@@ -0,0 +1,129 @@
+    email: false,
+    pw: false,
+  });
+  const [message, setMessage] = useState(false);
```
Set default value of message to `null`, so that message isn't assumed as bool type.

briannk replied:
Sort of unrelated but does React handle variable type internally? If I did what I did above and initialized as a boolean but later set it to say a string and passed it as a prop, would proptypes evaluate it based on initialization, so as a boolean still?

BryanHo10 commented on frontend/src/components/LogIn.js:
```diff
@@ -0,0 +1,129 @@
+    } else {
+      //submit
+      console.log("success");
+      const resp = await logIn(userInfo.email, userInfo.pw);
```
Wrap else logic into a separate function to reduce the size of the handleClick function. 
There's too many conditional statements. It makes it a bit hard to read. 

Likewise, think of a way to reduce the number of if/else blocks for the userInfo input validation. There's too many tracks to follow easily.

briannk replied:
What would you say is a good point to extract a block of code into a function? 5+ lines? I typically leave it as is if the logic relates to its containing function more than as a standalone function and if it doesn't really apply in any other scenario.

BryanHo10 commented on frontend/src/components/LogIn.js:
```diff
@@ -0,0 +1,129 @@
+      console.log("success");
+      const resp = await logIn(userInfo.email, userInfo.pw);
+      if (!isVerified) {
+        setMessage(
```
Instead of storing Message component, I would store content and change the state name to messageContent, so that the return value will be cleaner. 
Ex]
```
<Message content={messageContent}/>
```

BryanHo10 commented on frontend/src/components/LogIn.js:
```diff
@@ -0,0 +1,129 @@
+
+  return (
+    <div className={containerStyles}>
+      {message}
```
Reply Reference:: 
```
Instead of storing Message component, I would store content and change the state name to messageContent, so that the return value will be cleaner. 
```

BryanHo10 commented on frontend/src/components/PwReset.js:
```diff
@@ -0,0 +1,177 @@
+      {message}
+      {showResetForm ? (
+        isValidCode ? (
+          <>
```
Wrap all of these different views in a separate component with a descriptive name so that the return value will make sense at first glance.

briannk replied:
I removed those views as they were no longer necessary.

BryanHo10 commented on frontend/src/components/UserCard.js:
```diff
@@ -0,0 +1,55 @@
+    console.log(result);
+  };
+  return (
+    <div className={`${userCardStyles}`} id="user-card">
```
Why not just className={userCardStyles}?

briannk replied:
I initially had it like that in case I wanted to add another css class that was not in Tailwind for some of the Semantic components. I will change it back until the need arises in the future.

BryanHo10 commented on frontend/src/components/UserCard.js:
```diff
@@ -0,0 +1,55 @@
+  return (
+    <div className={`${userCardStyles}`} id="user-card">
+      {user && user.emailVerified ? (
+        <>
```
Wrap views into its own components to make things tidy.

briannk replied:
I'm not sure what you mean here. So instead of handling the render logic within this component I should have a component that renders the user card and a component that doesn't and return one of the two?

BryanHo10 replied:
Something like 
```
user && user.emailVerified ? <UserNav/>:<DefaultNav/>
```

briannk commented:
How would I go about implementing these changes if I had already modified some things then created a new branch to work on another feature? I changed some things after I pushed this branch and merged my updated branch with my local develop branch (since I couldn't pull from remote) then created a new branch to work on the feed feature.

BryanHo10 commented:
> How would I go about implementing these changes if I had already modified some things then created a new branch to work on another feature? I changed some things after I pushed this branch and merged my updated branch with my local develop branch (since I couldn't pull from remote) then created a new branch to work on the feed feature.

@briannk Ehh
I would suggest you clone the repo again and then once the changes have been done here, you can create a new branch with all of the changes you made from the old clone.
